{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [],
   "source": [
    "import pandas as pd\n",
    "import numpy as np\n",
    "import matplotlib.pyplot as plt \n",
    "%matplotlib inline\n",
    "import seaborn as sns\n",
    "sns.set(style=\"white\", color_codes=True)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Import of total overlap between Helix mtDB variants and candidate pathogenic variants from MitoMap\n",
    "This is the subset of the Helix mtDB bases that also have alleles in MitoMap\n",
    "\n",
    "**NOTE** although the same base has alleles in both datasets, there may be different alleles in each dataset per base\n",
    "\n",
    "- These entries need to be merged with the Helix mtDB (or a random sample of it) to add in candidate pathogenic mutations for model training and testing\n",
    "- There are 298 unique bases that appear in both the Helix mtDB and the MitoMap databases and 560 alleles are represented in the intersection\n",
    "- In the second code chunk below I eliminate most of the Helix mtDB alleles that do not appear in the MitoMap database"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>base</th>\n",
       "      <th>wildtype</th>\n",
       "      <th>gene</th>\n",
       "      <th>alleles</th>\n",
       "      <th>mutation</th>\n",
       "      <th>annotation</th>\n",
       "      <th>helix_feature</th>\n",
       "      <th>MM_allele</th>\n",
       "      <th>MM_feature</th>\n",
       "      <th>homoplasmic</th>\n",
       "      <th>heteroplasmic</th>\n",
       "      <th>counts_hom</th>\n",
       "      <th>AF_hom</th>\n",
       "      <th>counts_het</th>\n",
       "      <th>AF_het</th>\n",
       "      <th>pathogenic</th>\n",
       "      <th>disease</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>968</th>\n",
       "      <td>15942</td>\n",
       "      <td>T</td>\n",
       "      <td>MT-TT</td>\n",
       "      <td>[\"T\",\"C\"]</td>\n",
       "      <td>transition</td>\n",
       "      <td>NaN</td>\n",
       "      <td>tRNA_gene</td>\n",
       "      <td>T15942C</td>\n",
       "      <td>tRNA Thr</td>\n",
       "      <td>yes</td>\n",
       "      <td>yes</td>\n",
       "      <td>467</td>\n",
       "      <td>0.002383</td>\n",
       "      <td>7</td>\n",
       "      <td>0.000036</td>\n",
       "      <td>yes</td>\n",
       "      <td>Possibly LVNC-associated</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>969</th>\n",
       "      <td>15944</td>\n",
       "      <td>T</td>\n",
       "      <td>MT-TT</td>\n",
       "      <td>[\"T\",\"C\"]</td>\n",
       "      <td>transition</td>\n",
       "      <td>NaN</td>\n",
       "      <td>tRNA_gene</td>\n",
       "      <td>T15944del</td>\n",
       "      <td>tRNA Thr</td>\n",
       "      <td>yes</td>\n",
       "      <td>yes</td>\n",
       "      <td>12</td>\n",
       "      <td>0.000061</td>\n",
       "      <td>2</td>\n",
       "      <td>0.000010</td>\n",
       "      <td>yes</td>\n",
       "      <td>MM</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>970</th>\n",
       "      <td>15948</td>\n",
       "      <td>A</td>\n",
       "      <td>MT-TT</td>\n",
       "      <td>[\"A\",\"AGGACAAAT\"]</td>\n",
       "      <td>in-del</td>\n",
       "      <td>NaN</td>\n",
       "      <td>tRNA_gene</td>\n",
       "      <td>A15948G</td>\n",
       "      <td>tRNA Thr</td>\n",
       "      <td>yes</td>\n",
       "      <td>no</td>\n",
       "      <td>1</td>\n",
       "      <td>0.000005</td>\n",
       "      <td>0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>yes</td>\n",
       "      <td>Reported in tic disorder patient</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>971</th>\n",
       "      <td>15948</td>\n",
       "      <td>A</td>\n",
       "      <td>MT-TT</td>\n",
       "      <td>[\"A\",\"G\"]</td>\n",
       "      <td>transition</td>\n",
       "      <td>NaN</td>\n",
       "      <td>tRNA_gene</td>\n",
       "      <td>A15948G</td>\n",
       "      <td>tRNA Thr</td>\n",
       "      <td>yes</td>\n",
       "      <td>yes</td>\n",
       "      <td>3</td>\n",
       "      <td>0.000015</td>\n",
       "      <td>4</td>\n",
       "      <td>0.000020</td>\n",
       "      <td>yes</td>\n",
       "      <td>Reported in tic disorder patient</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>972</th>\n",
       "      <td>15950</td>\n",
       "      <td>G</td>\n",
       "      <td>MT-TT</td>\n",
       "      <td>[\"G\",\"A\"]</td>\n",
       "      <td>transition</td>\n",
       "      <td>NaN</td>\n",
       "      <td>tRNA_gene</td>\n",
       "      <td>G15950A</td>\n",
       "      <td>tRNA Thr</td>\n",
       "      <td>yes</td>\n",
       "      <td>yes</td>\n",
       "      <td>11</td>\n",
       "      <td>0.000056</td>\n",
       "      <td>2</td>\n",
       "      <td>0.000010</td>\n",
       "      <td>yes</td>\n",
       "      <td>Dopaminergic nerve cell death (PD) / tic disorder</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>973</th>\n",
       "      <td>15951</td>\n",
       "      <td>A</td>\n",
       "      <td>MT-TT</td>\n",
       "      <td>[\"A\",\"G\"]</td>\n",
       "      <td>transition</td>\n",
       "      <td>NaN</td>\n",
       "      <td>tRNA_gene</td>\n",
       "      <td>A15951G</td>\n",
       "      <td>tRNA Thr</td>\n",
       "      <td>yes</td>\n",
       "      <td>yes</td>\n",
       "      <td>208</td>\n",
       "      <td>0.001061</td>\n",
       "      <td>8</td>\n",
       "      <td>0.000041</td>\n",
       "      <td>yes</td>\n",
       "      <td>LHON / LHON modulator</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>974</th>\n",
       "      <td>15952</td>\n",
       "      <td>C</td>\n",
       "      <td>MT-TT</td>\n",
       "      <td>[\"C\",\"CA\"]</td>\n",
       "      <td>in-del</td>\n",
       "      <td>NaN</td>\n",
       "      <td>tRNA_gene</td>\n",
       "      <td>C15952T</td>\n",
       "      <td>tRNA Thr</td>\n",
       "      <td>yes</td>\n",
       "      <td>no</td>\n",
       "      <td>2</td>\n",
       "      <td>0.000010</td>\n",
       "      <td>0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>yes</td>\n",
       "      <td>Reported in tic disorder patient</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>975</th>\n",
       "      <td>15952</td>\n",
       "      <td>C</td>\n",
       "      <td>MT-TT</td>\n",
       "      <td>[\"C\",\"T\"]</td>\n",
       "      <td>transition</td>\n",
       "      <td>NaN</td>\n",
       "      <td>tRNA_gene</td>\n",
       "      <td>C15952T</td>\n",
       "      <td>tRNA Thr</td>\n",
       "      <td>yes</td>\n",
       "      <td>yes</td>\n",
       "      <td>6</td>\n",
       "      <td>0.000031</td>\n",
       "      <td>2</td>\n",
       "      <td>0.000010</td>\n",
       "      <td>yes</td>\n",
       "      <td>Reported in tic disorder patient</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>976</th>\n",
       "      <td>15952</td>\n",
       "      <td>C</td>\n",
       "      <td>MT-TT</td>\n",
       "      <td>[\"CA\",\"C\"]</td>\n",
       "      <td>in-del</td>\n",
       "      <td>NaN</td>\n",
       "      <td>tRNA_gene</td>\n",
       "      <td>C15952T</td>\n",
       "      <td>tRNA Thr</td>\n",
       "      <td>yes</td>\n",
       "      <td>no</td>\n",
       "      <td>6</td>\n",
       "      <td>0.000031</td>\n",
       "      <td>0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>yes</td>\n",
       "      <td>Reported in tic disorder patient</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>977</th>\n",
       "      <td>15965</td>\n",
       "      <td>A</td>\n",
       "      <td>MT-TP</td>\n",
       "      <td>[\"A\",\"G\"]</td>\n",
       "      <td>transition</td>\n",
       "      <td>NaN</td>\n",
       "      <td>tRNA_gene</td>\n",
       "      <td>A15965G</td>\n",
       "      <td>tRNA Pro</td>\n",
       "      <td>yes</td>\n",
       "      <td>no</td>\n",
       "      <td>116</td>\n",
       "      <td>0.000592</td>\n",
       "      <td>0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>yes</td>\n",
       "      <td>Dopaminergic nerve cell death (PD)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>978</th>\n",
       "      <td>15986</td>\n",
       "      <td>G</td>\n",
       "      <td>MT-TP</td>\n",
       "      <td>[\"G\",\"A\"]</td>\n",
       "      <td>transition</td>\n",
       "      <td>NaN</td>\n",
       "      <td>tRNA_gene</td>\n",
       "      <td>G15986GG</td>\n",
       "      <td>tRNA Pro</td>\n",
       "      <td>yes</td>\n",
       "      <td>yes</td>\n",
       "      <td>1</td>\n",
       "      <td>0.000005</td>\n",
       "      <td>7</td>\n",
       "      <td>0.000036</td>\n",
       "      <td>yes</td>\n",
       "      <td>LHON modulator</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>979</th>\n",
       "      <td>15992</td>\n",
       "      <td>A</td>\n",
       "      <td>MT-TP</td>\n",
       "      <td>[\"A\",\"G\"]</td>\n",
       "      <td>transition</td>\n",
       "      <td>NaN</td>\n",
       "      <td>tRNA_gene</td>\n",
       "      <td>A15992G</td>\n",
       "      <td>tRNA Pro</td>\n",
       "      <td>yes</td>\n",
       "      <td>yes</td>\n",
       "      <td>7</td>\n",
       "      <td>0.000036</td>\n",
       "      <td>2</td>\n",
       "      <td>0.000010</td>\n",
       "      <td>yes</td>\n",
       "      <td>Tic disorder patient / hearing loss patient</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>980</th>\n",
       "      <td>15992</td>\n",
       "      <td>A</td>\n",
       "      <td>MT-TP</td>\n",
       "      <td>[\"A\",\"G\"]</td>\n",
       "      <td>transition</td>\n",
       "      <td>NaN</td>\n",
       "      <td>tRNA_gene</td>\n",
       "      <td>A15992T</td>\n",
       "      <td>tRNA Pro</td>\n",
       "      <td>yes</td>\n",
       "      <td>yes</td>\n",
       "      <td>7</td>\n",
       "      <td>0.000036</td>\n",
       "      <td>2</td>\n",
       "      <td>0.000010</td>\n",
       "      <td>yes</td>\n",
       "      <td>Exercise-induced muscle swelling and fatigue</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>981</th>\n",
       "      <td>15995</td>\n",
       "      <td>G</td>\n",
       "      <td>MT-TP</td>\n",
       "      <td>[\"G\",\"A\"]</td>\n",
       "      <td>transition</td>\n",
       "      <td>NaN</td>\n",
       "      <td>tRNA_gene</td>\n",
       "      <td>G15995A</td>\n",
       "      <td>tRNA Pro</td>\n",
       "      <td>no</td>\n",
       "      <td>yes</td>\n",
       "      <td>0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>11</td>\n",
       "      <td>0.000056</td>\n",
       "      <td>yes</td>\n",
       "      <td>Mitochondrial cytopathy</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>982</th>\n",
       "      <td>15997</td>\n",
       "      <td>T</td>\n",
       "      <td>MT-TP</td>\n",
       "      <td>[\"T\",\"C\"]</td>\n",
       "      <td>transition</td>\n",
       "      <td>NaN</td>\n",
       "      <td>tRNA_gene</td>\n",
       "      <td>T15997C</td>\n",
       "      <td>tRNA Pro</td>\n",
       "      <td>no</td>\n",
       "      <td>yes</td>\n",
       "      <td>0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>2</td>\n",
       "      <td>0.000010</td>\n",
       "      <td>yes</td>\n",
       "      <td>Hearing loss patient</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>983</th>\n",
       "      <td>16002</td>\n",
       "      <td>T</td>\n",
       "      <td>MT-TP</td>\n",
       "      <td>[\"T\",\"C\"]</td>\n",
       "      <td>transition</td>\n",
       "      <td>NaN</td>\n",
       "      <td>tRNA_gene</td>\n",
       "      <td>T16002C</td>\n",
       "      <td>tRNA Pro</td>\n",
       "      <td>no</td>\n",
       "      <td>yes</td>\n",
       "      <td>0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>1</td>\n",
       "      <td>0.000005</td>\n",
       "      <td>yes</td>\n",
       "      <td>Mitochondrial cytopathy</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>984</th>\n",
       "      <td>16015</td>\n",
       "      <td>T</td>\n",
       "      <td>MT-TP</td>\n",
       "      <td>[\"T\",\"C\"]</td>\n",
       "      <td>transition</td>\n",
       "      <td>NaN</td>\n",
       "      <td>tRNA_gene</td>\n",
       "      <td>T16015C</td>\n",
       "      <td>tRNA Pro</td>\n",
       "      <td>no</td>\n",
       "      <td>yes</td>\n",
       "      <td>0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>3</td>\n",
       "      <td>0.000015</td>\n",
       "      <td>yes</td>\n",
       "      <td>Mitochondrial myopathy</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>985</th>\n",
       "      <td>16015</td>\n",
       "      <td>T</td>\n",
       "      <td>MT-TP</td>\n",
       "      <td>[\"T\",\"TATTCTCTGTTCTTTC\"]</td>\n",
       "      <td>in-del</td>\n",
       "      <td>NaN</td>\n",
       "      <td>tRNA_gene</td>\n",
       "      <td>T16015C</td>\n",
       "      <td>tRNA Pro</td>\n",
       "      <td>yes</td>\n",
       "      <td>no</td>\n",
       "      <td>2</td>\n",
       "      <td>0.000010</td>\n",
       "      <td>0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>yes</td>\n",
       "      <td>Mitochondrial myopathy</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>986</th>\n",
       "      <td>16018</td>\n",
       "      <td>T</td>\n",
       "      <td>MT-TP</td>\n",
       "      <td>[\"T\",\"C\"]</td>\n",
       "      <td>transition</td>\n",
       "      <td>NaN</td>\n",
       "      <td>tRNA_gene</td>\n",
       "      <td>T16018TTCTCTGTTCTTTCAT</td>\n",
       "      <td>tRNA Pro</td>\n",
       "      <td>yes</td>\n",
       "      <td>no</td>\n",
       "      <td>2</td>\n",
       "      <td>0.000010</td>\n",
       "      <td>0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>yes</td>\n",
       "      <td>Dilated cardiomyopathy (15 bp dup), alternate ...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>987</th>\n",
       "      <td>16023</td>\n",
       "      <td>G</td>\n",
       "      <td>MT-TP</td>\n",
       "      <td>[\"G\",\"A\"]</td>\n",
       "      <td>transition</td>\n",
       "      <td>NaN</td>\n",
       "      <td>tRNA_gene</td>\n",
       "      <td>G16023A</td>\n",
       "      <td>tRNA Pro</td>\n",
       "      <td>no</td>\n",
       "      <td>yes</td>\n",
       "      <td>0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>3</td>\n",
       "      <td>0.000015</td>\n",
       "      <td>yes</td>\n",
       "      <td>Migraine +pigmentary retinopathy +deafness +le...</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "      base wildtype   gene                   alleles    mutation annotation  \\\n",
       "968  15942        T  MT-TT                 [\"T\",\"C\"]  transition        NaN   \n",
       "969  15944        T  MT-TT                 [\"T\",\"C\"]  transition        NaN   \n",
       "970  15948        A  MT-TT         [\"A\",\"AGGACAAAT\"]      in-del        NaN   \n",
       "971  15948        A  MT-TT                 [\"A\",\"G\"]  transition        NaN   \n",
       "972  15950        G  MT-TT                 [\"G\",\"A\"]  transition        NaN   \n",
       "973  15951        A  MT-TT                 [\"A\",\"G\"]  transition        NaN   \n",
       "974  15952        C  MT-TT                [\"C\",\"CA\"]      in-del        NaN   \n",
       "975  15952        C  MT-TT                 [\"C\",\"T\"]  transition        NaN   \n",
       "976  15952        C  MT-TT                [\"CA\",\"C\"]      in-del        NaN   \n",
       "977  15965        A  MT-TP                 [\"A\",\"G\"]  transition        NaN   \n",
       "978  15986        G  MT-TP                 [\"G\",\"A\"]  transition        NaN   \n",
       "979  15992        A  MT-TP                 [\"A\",\"G\"]  transition        NaN   \n",
       "980  15992        A  MT-TP                 [\"A\",\"G\"]  transition        NaN   \n",
       "981  15995        G  MT-TP                 [\"G\",\"A\"]  transition        NaN   \n",
       "982  15997        T  MT-TP                 [\"T\",\"C\"]  transition        NaN   \n",
       "983  16002        T  MT-TP                 [\"T\",\"C\"]  transition        NaN   \n",
       "984  16015        T  MT-TP                 [\"T\",\"C\"]  transition        NaN   \n",
       "985  16015        T  MT-TP  [\"T\",\"TATTCTCTGTTCTTTC\"]      in-del        NaN   \n",
       "986  16018        T  MT-TP                 [\"T\",\"C\"]  transition        NaN   \n",
       "987  16023        G  MT-TP                 [\"G\",\"A\"]  transition        NaN   \n",
       "\n",
       "    helix_feature               MM_allele MM_feature homoplasmic  \\\n",
       "968     tRNA_gene                 T15942C   tRNA Thr         yes   \n",
       "969     tRNA_gene               T15944del   tRNA Thr         yes   \n",
       "970     tRNA_gene                 A15948G   tRNA Thr         yes   \n",
       "971     tRNA_gene                 A15948G   tRNA Thr         yes   \n",
       "972     tRNA_gene                 G15950A   tRNA Thr         yes   \n",
       "973     tRNA_gene                 A15951G   tRNA Thr         yes   \n",
       "974     tRNA_gene                 C15952T   tRNA Thr         yes   \n",
       "975     tRNA_gene                 C15952T   tRNA Thr         yes   \n",
       "976     tRNA_gene                 C15952T   tRNA Thr         yes   \n",
       "977     tRNA_gene                 A15965G   tRNA Pro         yes   \n",
       "978     tRNA_gene                G15986GG   tRNA Pro         yes   \n",
       "979     tRNA_gene                 A15992G   tRNA Pro         yes   \n",
       "980     tRNA_gene                 A15992T   tRNA Pro         yes   \n",
       "981     tRNA_gene                 G15995A   tRNA Pro          no   \n",
       "982     tRNA_gene                 T15997C   tRNA Pro          no   \n",
       "983     tRNA_gene                 T16002C   tRNA Pro          no   \n",
       "984     tRNA_gene                 T16015C   tRNA Pro          no   \n",
       "985     tRNA_gene                 T16015C   tRNA Pro         yes   \n",
       "986     tRNA_gene  T16018TTCTCTGTTCTTTCAT   tRNA Pro         yes   \n",
       "987     tRNA_gene                 G16023A   tRNA Pro          no   \n",
       "\n",
       "    heteroplasmic  counts_hom    AF_hom  counts_het    AF_het pathogenic  \\\n",
       "968           yes         467  0.002383           7  0.000036        yes   \n",
       "969           yes          12  0.000061           2  0.000010        yes   \n",
       "970            no           1  0.000005           0  0.000000        yes   \n",
       "971           yes           3  0.000015           4  0.000020        yes   \n",
       "972           yes          11  0.000056           2  0.000010        yes   \n",
       "973           yes         208  0.001061           8  0.000041        yes   \n",
       "974            no           2  0.000010           0  0.000000        yes   \n",
       "975           yes           6  0.000031           2  0.000010        yes   \n",
       "976            no           6  0.000031           0  0.000000        yes   \n",
       "977            no         116  0.000592           0  0.000000        yes   \n",
       "978           yes           1  0.000005           7  0.000036        yes   \n",
       "979           yes           7  0.000036           2  0.000010        yes   \n",
       "980           yes           7  0.000036           2  0.000010        yes   \n",
       "981           yes           0  0.000000          11  0.000056        yes   \n",
       "982           yes           0  0.000000           2  0.000010        yes   \n",
       "983           yes           0  0.000000           1  0.000005        yes   \n",
       "984           yes           0  0.000000           3  0.000015        yes   \n",
       "985            no           2  0.000010           0  0.000000        yes   \n",
       "986            no           2  0.000010           0  0.000000        yes   \n",
       "987           yes           0  0.000000           3  0.000015        yes   \n",
       "\n",
       "                                               disease  \n",
       "968                           Possibly LVNC-associated  \n",
       "969                                                 MM  \n",
       "970                   Reported in tic disorder patient  \n",
       "971                   Reported in tic disorder patient  \n",
       "972  Dopaminergic nerve cell death (PD) / tic disorder  \n",
       "973                              LHON / LHON modulator  \n",
       "974                   Reported in tic disorder patient  \n",
       "975                   Reported in tic disorder patient  \n",
       "976                   Reported in tic disorder patient  \n",
       "977                 Dopaminergic nerve cell death (PD)  \n",
       "978                                     LHON modulator  \n",
       "979        Tic disorder patient / hearing loss patient  \n",
       "980       Exercise-induced muscle swelling and fatigue  \n",
       "981                            Mitochondrial cytopathy  \n",
       "982                               Hearing loss patient  \n",
       "983                            Mitochondrial cytopathy  \n",
       "984                             Mitochondrial myopathy  \n",
       "985                             Mitochondrial myopathy  \n",
       "986  Dilated cardiomyopathy (15 bp dup), alternate ...  \n",
       "987  Migraine +pigmentary retinopathy +deafness +le...  "
      ]
     },
     "execution_count": 3,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "totHelMitoVars = pd.read_csv(\"/Users/leekapp/Desktop/CS254/Assignments/FinalProject/totHelMitoVars.csv\")\n",
    "totHelMitoVars.tail(20)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Limiting overlap dataset to unique bases\n",
    "The following process limits the overlap database to Helix mtDB alleles that match the MitoMap alleles because most bases have several alleles in the Helix database but only one pathogenic allele for that base exists in MitoMap"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "C-T\n",
      "T\n",
      "[\"C\",\"A\"]\n",
      "A\n"
     ]
    }
   ],
   "source": [
    "# checking out how to access specific bases of alleles\n",
    "print(totHelMitoVars['MM_allele'][0]) \n",
    "print(totHelMitoVars['MM_allele'][0][-1])\n",
    "print(totHelMitoVars['alleles'][0])\n",
    "print(totHelMitoVars['alleles'][0][6])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [],
   "source": [
    "# creating 'first' and 'last' columns to match Helix allele (last base) with last base of MM allele\n",
    "idx = totHelMitoVars.index\n",
    "first = np.full(idx.size, 'x')\n",
    "for i in range(len(idx)):\n",
    "    first[i] = totHelMitoVars['alleles'][i][-3]\n",
    "    #first[i] = int(first[i])\n",
    "first = pd.Series(first)\n",
    "\n",
    "last = np.full(idx.size, 'x')\n",
    "for i in range(len(idx)):\n",
    "    last[i] = totHelMitoVars['MM_allele'][i][-1]\n",
    "    #first[i] = int(first[i])\n",
    "last = pd.Series(last)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>base</th>\n",
       "      <th>first</th>\n",
       "      <th>last</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>114</td>\n",
       "      <td>A</td>\n",
       "      <td>T</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>114</td>\n",
       "      <td>G</td>\n",
       "      <td>T</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>114</td>\n",
       "      <td>T</td>\n",
       "      <td>T</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>114</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>146</td>\n",
       "      <td>A</td>\n",
       "      <td>C</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>983</th>\n",
       "      <td>16002</td>\n",
       "      <td>C</td>\n",
       "      <td>C</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>984</th>\n",
       "      <td>16015</td>\n",
       "      <td>C</td>\n",
       "      <td>C</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>985</th>\n",
       "      <td>16015</td>\n",
       "      <td>C</td>\n",
       "      <td>C</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>986</th>\n",
       "      <td>16018</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>987</th>\n",
       "      <td>16023</td>\n",
       "      <td>A</td>\n",
       "      <td>A</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>988 rows × 3 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "      base first last\n",
       "0      114     A    T\n",
       "1      114     G    T\n",
       "2      114     T    T\n",
       "3      114     C    T\n",
       "4      146     A    C\n",
       "..     ...   ...  ...\n",
       "983  16002     C    C\n",
       "984  16015     C    C\n",
       "985  16015     C    C\n",
       "986  16018     C    T\n",
       "987  16023     A    A\n",
       "\n",
       "[988 rows x 3 columns]"
      ]
     },
     "execution_count": 7,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# matching base number with alleles\n",
    "nucleotides = pd.DataFrame({'base' : totHelMitoVars['base'], 'first': first, 'last': last})\n",
    "nucleotides"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "Int64Index([  2,   5,   8,   9,  10,  16,  17,  19,  20,  23,\n",
       "            ...\n",
       "            973, 975, 977, 979, 981, 982, 983, 984, 985, 987],\n",
       "           dtype='int64', length=574)"
      ]
     },
     "execution_count": 8,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "index = nucleotides.loc[nucleotides['first'] == nucleotides['last']] #gives rows where alleles match by last base of Helix allele\n",
    "index.index #gives just the value of the row index to use to index the overlap dataframe"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### These data still have duplicates for matched MitoMap and Helix mtDB bases"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>base</th>\n",
       "      <th>wildtype</th>\n",
       "      <th>gene</th>\n",
       "      <th>alleles</th>\n",
       "      <th>mutation</th>\n",
       "      <th>annotation</th>\n",
       "      <th>helix_feature</th>\n",
       "      <th>MM_allele</th>\n",
       "      <th>MM_feature</th>\n",
       "      <th>homoplasmic</th>\n",
       "      <th>heteroplasmic</th>\n",
       "      <th>counts_hom</th>\n",
       "      <th>AF_hom</th>\n",
       "      <th>counts_het</th>\n",
       "      <th>AF_het</th>\n",
       "      <th>pathogenic</th>\n",
       "      <th>disease</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>114</td>\n",
       "      <td>C</td>\n",
       "      <td>MT-CRb</td>\n",
       "      <td>[\"C\",\"T\"]</td>\n",
       "      <td>transition</td>\n",
       "      <td>NaN</td>\n",
       "      <td>non_coding</td>\n",
       "      <td>C-T</td>\n",
       "      <td>noncoding</td>\n",
       "      <td>yes</td>\n",
       "      <td>yes</td>\n",
       "      <td>1974</td>\n",
       "      <td>0.010072</td>\n",
       "      <td>101</td>\n",
       "      <td>0.000515</td>\n",
       "      <td>yes</td>\n",
       "      <td>BD-associated</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>146</td>\n",
       "      <td>T</td>\n",
       "      <td>MT-CRb</td>\n",
       "      <td>[\"T\",\"C\"]</td>\n",
       "      <td>transition</td>\n",
       "      <td>NaN</td>\n",
       "      <td>non_coding</td>\n",
       "      <td>T-C</td>\n",
       "      <td>noncoding</td>\n",
       "      <td>yes</td>\n",
       "      <td>yes</td>\n",
       "      <td>25525</td>\n",
       "      <td>0.130241</td>\n",
       "      <td>622</td>\n",
       "      <td>0.003174</td>\n",
       "      <td>yes</td>\n",
       "      <td>Absence of Endometriosis</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>150</td>\n",
       "      <td>C</td>\n",
       "      <td>MT-CRb</td>\n",
       "      <td>[\"C\",\"T\"]</td>\n",
       "      <td>transition</td>\n",
       "      <td>NaN</td>\n",
       "      <td>non_coding</td>\n",
       "      <td>C-T</td>\n",
       "      <td>noncoding</td>\n",
       "      <td>yes</td>\n",
       "      <td>yes</td>\n",
       "      <td>19751</td>\n",
       "      <td>0.100779</td>\n",
       "      <td>168</td>\n",
       "      <td>0.000857</td>\n",
       "      <td>yes</td>\n",
       "      <td>Longevity / Cervical Carcinoma / HPV infection...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>185</td>\n",
       "      <td>G</td>\n",
       "      <td>MT-CRb</td>\n",
       "      <td>[\"G\",\"A\"]</td>\n",
       "      <td>transition</td>\n",
       "      <td>NaN</td>\n",
       "      <td>non_coding</td>\n",
       "      <td>G-A</td>\n",
       "      <td>noncoding</td>\n",
       "      <td>yes</td>\n",
       "      <td>yes</td>\n",
       "      <td>10568</td>\n",
       "      <td>0.053923</td>\n",
       "      <td>317</td>\n",
       "      <td>0.001617</td>\n",
       "      <td>yes</td>\n",
       "      <td>Low VO2max response</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>185</td>\n",
       "      <td>G</td>\n",
       "      <td>MT-CRb</td>\n",
       "      <td>[\"G\",\"A\",\"GAA\"]</td>\n",
       "      <td>in-del</td>\n",
       "      <td>NaN</td>\n",
       "      <td>non_coding</td>\n",
       "      <td>G-A</td>\n",
       "      <td>noncoding</td>\n",
       "      <td>no</td>\n",
       "      <td>yes</td>\n",
       "      <td>0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>1</td>\n",
       "      <td>0.000005</td>\n",
       "      <td>yes</td>\n",
       "      <td>Low VO2max response</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16</th>\n",
       "      <td>195</td>\n",
       "      <td>T</td>\n",
       "      <td>MT-CRb</td>\n",
       "      <td>[\"T\",\"A\",\"C\"]</td>\n",
       "      <td>in-del</td>\n",
       "      <td>NaN</td>\n",
       "      <td>non_coding</td>\n",
       "      <td>T-C</td>\n",
       "      <td>noncoding</td>\n",
       "      <td>no</td>\n",
       "      <td>yes</td>\n",
       "      <td>0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>2</td>\n",
       "      <td>0.000010</td>\n",
       "      <td>yes</td>\n",
       "      <td>BD-associated / melanoma pts</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17</th>\n",
       "      <td>195</td>\n",
       "      <td>T</td>\n",
       "      <td>MT-CRb</td>\n",
       "      <td>[\"T\",\"C\"]</td>\n",
       "      <td>transition</td>\n",
       "      <td>NaN</td>\n",
       "      <td>non_coding</td>\n",
       "      <td>T-C</td>\n",
       "      <td>noncoding</td>\n",
       "      <td>yes</td>\n",
       "      <td>yes</td>\n",
       "      <td>34496</td>\n",
       "      <td>0.176015</td>\n",
       "      <td>1050</td>\n",
       "      <td>0.005358</td>\n",
       "      <td>yes</td>\n",
       "      <td>BD-associated / melanoma pts</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19</th>\n",
       "      <td>228</td>\n",
       "      <td>G</td>\n",
       "      <td>MT-CRb</td>\n",
       "      <td>[\"G\",\"A\"]</td>\n",
       "      <td>transition</td>\n",
       "      <td>NaN</td>\n",
       "      <td>non_coding</td>\n",
       "      <td>G-A</td>\n",
       "      <td>noncoding</td>\n",
       "      <td>yes</td>\n",
       "      <td>yes</td>\n",
       "      <td>9673</td>\n",
       "      <td>0.049356</td>\n",
       "      <td>191</td>\n",
       "      <td>0.000975</td>\n",
       "      <td>yes</td>\n",
       "      <td>Low VO2max response</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20</th>\n",
       "      <td>228</td>\n",
       "      <td>G</td>\n",
       "      <td>MT-CRb</td>\n",
       "      <td>[\"G\",\"A\",\"GACATAATAA\"]</td>\n",
       "      <td>in-del</td>\n",
       "      <td>NaN</td>\n",
       "      <td>non_coding</td>\n",
       "      <td>G-A</td>\n",
       "      <td>noncoding</td>\n",
       "      <td>no</td>\n",
       "      <td>yes</td>\n",
       "      <td>0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>1</td>\n",
       "      <td>0.000005</td>\n",
       "      <td>yes</td>\n",
       "      <td>Low VO2max response</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>23</th>\n",
       "      <td>295</td>\n",
       "      <td>C</td>\n",
       "      <td>MT-CRb</td>\n",
       "      <td>[\"C\",\"T\"]</td>\n",
       "      <td>transition</td>\n",
       "      <td>NaN</td>\n",
       "      <td>non_coding</td>\n",
       "      <td>C-T</td>\n",
       "      <td>noncoding</td>\n",
       "      <td>yes</td>\n",
       "      <td>yes</td>\n",
       "      <td>15983</td>\n",
       "      <td>0.081553</td>\n",
       "      <td>8</td>\n",
       "      <td>0.000041</td>\n",
       "      <td>yes</td>\n",
       "      <td>Low VO2max response</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "    base wildtype    gene                 alleles    mutation annotation  \\\n",
       "2    114        C  MT-CRb               [\"C\",\"T\"]  transition        NaN   \n",
       "5    146        T  MT-CRb               [\"T\",\"C\"]  transition        NaN   \n",
       "8    150        C  MT-CRb               [\"C\",\"T\"]  transition        NaN   \n",
       "9    185        G  MT-CRb               [\"G\",\"A\"]  transition        NaN   \n",
       "10   185        G  MT-CRb         [\"G\",\"A\",\"GAA\"]      in-del        NaN   \n",
       "16   195        T  MT-CRb           [\"T\",\"A\",\"C\"]      in-del        NaN   \n",
       "17   195        T  MT-CRb               [\"T\",\"C\"]  transition        NaN   \n",
       "19   228        G  MT-CRb               [\"G\",\"A\"]  transition        NaN   \n",
       "20   228        G  MT-CRb  [\"G\",\"A\",\"GACATAATAA\"]      in-del        NaN   \n",
       "23   295        C  MT-CRb               [\"C\",\"T\"]  transition        NaN   \n",
       "\n",
       "   helix_feature MM_allele MM_feature homoplasmic heteroplasmic  counts_hom  \\\n",
       "2     non_coding       C-T  noncoding         yes           yes        1974   \n",
       "5     non_coding       T-C  noncoding         yes           yes       25525   \n",
       "8     non_coding       C-T  noncoding         yes           yes       19751   \n",
       "9     non_coding       G-A  noncoding         yes           yes       10568   \n",
       "10    non_coding       G-A  noncoding          no           yes           0   \n",
       "16    non_coding       T-C  noncoding          no           yes           0   \n",
       "17    non_coding       T-C  noncoding         yes           yes       34496   \n",
       "19    non_coding       G-A  noncoding         yes           yes        9673   \n",
       "20    non_coding       G-A  noncoding          no           yes           0   \n",
       "23    non_coding       C-T  noncoding         yes           yes       15983   \n",
       "\n",
       "      AF_hom  counts_het    AF_het pathogenic  \\\n",
       "2   0.010072         101  0.000515        yes   \n",
       "5   0.130241         622  0.003174        yes   \n",
       "8   0.100779         168  0.000857        yes   \n",
       "9   0.053923         317  0.001617        yes   \n",
       "10  0.000000           1  0.000005        yes   \n",
       "16  0.000000           2  0.000010        yes   \n",
       "17  0.176015        1050  0.005358        yes   \n",
       "19  0.049356         191  0.000975        yes   \n",
       "20  0.000000           1  0.000005        yes   \n",
       "23  0.081553           8  0.000041        yes   \n",
       "\n",
       "                                              disease  \n",
       "2                                       BD-associated  \n",
       "5                            Absence of Endometriosis  \n",
       "8   Longevity / Cervical Carcinoma / HPV infection...  \n",
       "9                                 Low VO2max response  \n",
       "10                                Low VO2max response  \n",
       "16                       BD-associated / melanoma pts  \n",
       "17                       BD-associated / melanoma pts  \n",
       "19                                Low VO2max response  \n",
       "20                                Low VO2max response  \n",
       "23                                Low VO2max response  "
      ]
     },
     "execution_count": 9,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "uniqueOverlaps = totHelMitoVars.iloc[index.index,:]\n",
    "uniqueOverlaps.head(10)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "### Finding the duplicated bases"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>base</th>\n",
       "      <th>wildtype</th>\n",
       "      <th>gene</th>\n",
       "      <th>alleles</th>\n",
       "      <th>mutation</th>\n",
       "      <th>annotation</th>\n",
       "      <th>helix_feature</th>\n",
       "      <th>MM_allele</th>\n",
       "      <th>MM_feature</th>\n",
       "      <th>homoplasmic</th>\n",
       "      <th>heteroplasmic</th>\n",
       "      <th>counts_hom</th>\n",
       "      <th>AF_hom</th>\n",
       "      <th>counts_het</th>\n",
       "      <th>AF_het</th>\n",
       "      <th>pathogenic</th>\n",
       "      <th>disease</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>185</td>\n",
       "      <td>G</td>\n",
       "      <td>MT-CRb</td>\n",
       "      <td>[\"G\",\"A\"]</td>\n",
       "      <td>transition</td>\n",
       "      <td>NaN</td>\n",
       "      <td>non_coding</td>\n",
       "      <td>G-A</td>\n",
       "      <td>noncoding</td>\n",
       "      <td>yes</td>\n",
       "      <td>yes</td>\n",
       "      <td>10568</td>\n",
       "      <td>0.053923</td>\n",
       "      <td>317</td>\n",
       "      <td>0.001617</td>\n",
       "      <td>yes</td>\n",
       "      <td>Low VO2max response</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>185</td>\n",
       "      <td>G</td>\n",
       "      <td>MT-CRb</td>\n",
       "      <td>[\"G\",\"A\",\"GAA\"]</td>\n",
       "      <td>in-del</td>\n",
       "      <td>NaN</td>\n",
       "      <td>non_coding</td>\n",
       "      <td>G-A</td>\n",
       "      <td>noncoding</td>\n",
       "      <td>no</td>\n",
       "      <td>yes</td>\n",
       "      <td>0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>1</td>\n",
       "      <td>0.000005</td>\n",
       "      <td>yes</td>\n",
       "      <td>Low VO2max response</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16</th>\n",
       "      <td>195</td>\n",
       "      <td>T</td>\n",
       "      <td>MT-CRb</td>\n",
       "      <td>[\"T\",\"A\",\"C\"]</td>\n",
       "      <td>in-del</td>\n",
       "      <td>NaN</td>\n",
       "      <td>non_coding</td>\n",
       "      <td>T-C</td>\n",
       "      <td>noncoding</td>\n",
       "      <td>no</td>\n",
       "      <td>yes</td>\n",
       "      <td>0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>2</td>\n",
       "      <td>0.000010</td>\n",
       "      <td>yes</td>\n",
       "      <td>BD-associated / melanoma pts</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17</th>\n",
       "      <td>195</td>\n",
       "      <td>T</td>\n",
       "      <td>MT-CRb</td>\n",
       "      <td>[\"T\",\"C\"]</td>\n",
       "      <td>transition</td>\n",
       "      <td>NaN</td>\n",
       "      <td>non_coding</td>\n",
       "      <td>T-C</td>\n",
       "      <td>noncoding</td>\n",
       "      <td>yes</td>\n",
       "      <td>yes</td>\n",
       "      <td>34496</td>\n",
       "      <td>0.176015</td>\n",
       "      <td>1050</td>\n",
       "      <td>0.005358</td>\n",
       "      <td>yes</td>\n",
       "      <td>BD-associated / melanoma pts</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19</th>\n",
       "      <td>228</td>\n",
       "      <td>G</td>\n",
       "      <td>MT-CRb</td>\n",
       "      <td>[\"G\",\"A\"]</td>\n",
       "      <td>transition</td>\n",
       "      <td>NaN</td>\n",
       "      <td>non_coding</td>\n",
       "      <td>G-A</td>\n",
       "      <td>noncoding</td>\n",
       "      <td>yes</td>\n",
       "      <td>yes</td>\n",
       "      <td>9673</td>\n",
       "      <td>0.049356</td>\n",
       "      <td>191</td>\n",
       "      <td>0.000975</td>\n",
       "      <td>yes</td>\n",
       "      <td>Low VO2max response</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20</th>\n",
       "      <td>228</td>\n",
       "      <td>G</td>\n",
       "      <td>MT-CRb</td>\n",
       "      <td>[\"G\",\"A\",\"GACATAATAA\"]</td>\n",
       "      <td>in-del</td>\n",
       "      <td>NaN</td>\n",
       "      <td>non_coding</td>\n",
       "      <td>G-A</td>\n",
       "      <td>noncoding</td>\n",
       "      <td>no</td>\n",
       "      <td>yes</td>\n",
       "      <td>0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>1</td>\n",
       "      <td>0.000005</td>\n",
       "      <td>yes</td>\n",
       "      <td>Low VO2max response</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>38</th>\n",
       "      <td>576</td>\n",
       "      <td>A</td>\n",
       "      <td>MT-CRb</td>\n",
       "      <td>[\"A\",\"G\"]</td>\n",
       "      <td>transition</td>\n",
       "      <td>NaN</td>\n",
       "      <td>non_coding</td>\n",
       "      <td>A-G</td>\n",
       "      <td>noncoding MT-TF precursor</td>\n",
       "      <td>yes</td>\n",
       "      <td>yes</td>\n",
       "      <td>10</td>\n",
       "      <td>0.000051</td>\n",
       "      <td>1</td>\n",
       "      <td>0.000005</td>\n",
       "      <td>yes</td>\n",
       "      <td>Hearing loss patient</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>40</th>\n",
       "      <td>3308</td>\n",
       "      <td>T</td>\n",
       "      <td>MT-ND1</td>\n",
       "      <td>[\"T\",\"C\"]</td>\n",
       "      <td>transition</td>\n",
       "      <td>2</td>\n",
       "      <td>protein_coding_gene</td>\n",
       "      <td>T-C</td>\n",
       "      <td>M1T</td>\n",
       "      <td>yes</td>\n",
       "      <td>yes</td>\n",
       "      <td>1126</td>\n",
       "      <td>0.005745</td>\n",
       "      <td>26</td>\n",
       "      <td>0.000133</td>\n",
       "      <td>yes</td>\n",
       "      <td>MELAS / DEAF enhancer / hypertension / LVNC / ...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>41</th>\n",
       "      <td>3308</td>\n",
       "      <td>T</td>\n",
       "      <td>MT-ND1</td>\n",
       "      <td>[\"T\",\"G\"]</td>\n",
       "      <td>transversion</td>\n",
       "      <td>2</td>\n",
       "      <td>protein_coding_gene</td>\n",
       "      <td>T-G</td>\n",
       "      <td>M1term</td>\n",
       "      <td>yes</td>\n",
       "      <td>no</td>\n",
       "      <td>89</td>\n",
       "      <td>0.000454</td>\n",
       "      <td>0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>yes</td>\n",
       "      <td>Sudden Infant Death</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>176</th>\n",
       "      <td>7445</td>\n",
       "      <td>A</td>\n",
       "      <td>MT-CO1</td>\n",
       "      <td>[\"A\",\"C\"]</td>\n",
       "      <td>transversion</td>\n",
       "      <td>3</td>\n",
       "      <td>protein_coding_gene</td>\n",
       "      <td>A-C</td>\n",
       "      <td>term514S</td>\n",
       "      <td>yes</td>\n",
       "      <td>no</td>\n",
       "      <td>1</td>\n",
       "      <td>0.000005</td>\n",
       "      <td>0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>yes</td>\n",
       "      <td>DEAF</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>179</th>\n",
       "      <td>7445</td>\n",
       "      <td>A</td>\n",
       "      <td>MT-CO1</td>\n",
       "      <td>[\"A\",\"G\"]</td>\n",
       "      <td>transition</td>\n",
       "      <td>3</td>\n",
       "      <td>protein_coding_gene</td>\n",
       "      <td>A-G</td>\n",
       "      <td>term514term</td>\n",
       "      <td>no</td>\n",
       "      <td>yes</td>\n",
       "      <td>0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>6</td>\n",
       "      <td>0.000031</td>\n",
       "      <td>yes</td>\n",
       "      <td>SNHL</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>266</th>\n",
       "      <td>8993</td>\n",
       "      <td>T</td>\n",
       "      <td>MT-ATP6</td>\n",
       "      <td>[\"T\",\"C\"]</td>\n",
       "      <td>transition</td>\n",
       "      <td>2</td>\n",
       "      <td>protein_coding_gene</td>\n",
       "      <td>T-C</td>\n",
       "      <td>L156P</td>\n",
       "      <td>no</td>\n",
       "      <td>yes</td>\n",
       "      <td>0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>2</td>\n",
       "      <td>0.000010</td>\n",
       "      <td>yes</td>\n",
       "      <td>NARP / Leigh Disease / MILS / other</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>269</th>\n",
       "      <td>8993</td>\n",
       "      <td>T</td>\n",
       "      <td>MT-ATP6</td>\n",
       "      <td>[\"T\",\"G\"]</td>\n",
       "      <td>transversion</td>\n",
       "      <td>2</td>\n",
       "      <td>protein_coding_gene</td>\n",
       "      <td>T-G</td>\n",
       "      <td>L156R</td>\n",
       "      <td>no</td>\n",
       "      <td>yes</td>\n",
       "      <td>0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>1</td>\n",
       "      <td>0.000005</td>\n",
       "      <td>yes</td>\n",
       "      <td>NARP / Leigh Disease / MILS / other</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>532</th>\n",
       "      <td>576</td>\n",
       "      <td>A</td>\n",
       "      <td>MT-CRb</td>\n",
       "      <td>[\"A\",\"G\"]</td>\n",
       "      <td>transition</td>\n",
       "      <td>NaN</td>\n",
       "      <td>non_coding</td>\n",
       "      <td>A576G</td>\n",
       "      <td>noncoding MT-TF precursor</td>\n",
       "      <td>yes</td>\n",
       "      <td>yes</td>\n",
       "      <td>10</td>\n",
       "      <td>0.000051</td>\n",
       "      <td>1</td>\n",
       "      <td>0.000005</td>\n",
       "      <td>yes</td>\n",
       "      <td>Hearing loss patient</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>534</th>\n",
       "      <td>593</td>\n",
       "      <td>T</td>\n",
       "      <td>MT-TF</td>\n",
       "      <td>[\"T\",\"C\"]</td>\n",
       "      <td>transition</td>\n",
       "      <td>NaN</td>\n",
       "      <td>tRNA_gene</td>\n",
       "      <td>T593C</td>\n",
       "      <td>tRNA Phe</td>\n",
       "      <td>yes</td>\n",
       "      <td>yes</td>\n",
       "      <td>480</td>\n",
       "      <td>0.002449</td>\n",
       "      <td>41</td>\n",
       "      <td>0.000209</td>\n",
       "      <td>yes</td>\n",
       "      <td>Nonsyndromic hearing loss</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>536</th>\n",
       "      <td>593</td>\n",
       "      <td>T</td>\n",
       "      <td>MT-TF</td>\n",
       "      <td>[\"T\",\"TC\"]</td>\n",
       "      <td>in-del</td>\n",
       "      <td>NaN</td>\n",
       "      <td>tRNA_gene</td>\n",
       "      <td>T593C</td>\n",
       "      <td>tRNA Phe</td>\n",
       "      <td>yes</td>\n",
       "      <td>no</td>\n",
       "      <td>86</td>\n",
       "      <td>0.000439</td>\n",
       "      <td>0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>yes</td>\n",
       "      <td>Nonsyndromic hearing loss</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>557</th>\n",
       "      <td>745</td>\n",
       "      <td>A</td>\n",
       "      <td>MT-RNR1</td>\n",
       "      <td>[\"A\",\"AG\"]</td>\n",
       "      <td>in-del</td>\n",
       "      <td>NaN</td>\n",
       "      <td>rRNA_gene</td>\n",
       "      <td>A745G</td>\n",
       "      <td>12S rRNA</td>\n",
       "      <td>no</td>\n",
       "      <td>yes</td>\n",
       "      <td>0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>1</td>\n",
       "      <td>0.000005</td>\n",
       "      <td>yes</td>\n",
       "      <td>DEAF-associated</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>559</th>\n",
       "      <td>745</td>\n",
       "      <td>A</td>\n",
       "      <td>MT-RNR1</td>\n",
       "      <td>[\"A\",\"G\"]</td>\n",
       "      <td>transition</td>\n",
       "      <td>NaN</td>\n",
       "      <td>rRNA_gene</td>\n",
       "      <td>A745G</td>\n",
       "      <td>12S rRNA</td>\n",
       "      <td>yes</td>\n",
       "      <td>yes</td>\n",
       "      <td>333</td>\n",
       "      <td>0.001699</td>\n",
       "      <td>3</td>\n",
       "      <td>0.000015</td>\n",
       "      <td>yes</td>\n",
       "      <td>DEAF-associated</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>561</th>\n",
       "      <td>792</td>\n",
       "      <td>C</td>\n",
       "      <td>MT-RNR1</td>\n",
       "      <td>[\"C\",\"CT\"]</td>\n",
       "      <td>in-del</td>\n",
       "      <td>NaN</td>\n",
       "      <td>rRNA_gene</td>\n",
       "      <td>C792T</td>\n",
       "      <td>12S rRNA</td>\n",
       "      <td>yes</td>\n",
       "      <td>no</td>\n",
       "      <td>7</td>\n",
       "      <td>0.000036</td>\n",
       "      <td>0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>yes</td>\n",
       "      <td>Increased risk of nonsyndromic deafness</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>562</th>\n",
       "      <td>792</td>\n",
       "      <td>C</td>\n",
       "      <td>MT-RNR1</td>\n",
       "      <td>[\"C\",\"T\"]</td>\n",
       "      <td>transition</td>\n",
       "      <td>NaN</td>\n",
       "      <td>rRNA_gene</td>\n",
       "      <td>C792T</td>\n",
       "      <td>12S rRNA</td>\n",
       "      <td>yes</td>\n",
       "      <td>yes</td>\n",
       "      <td>11</td>\n",
       "      <td>0.000056</td>\n",
       "      <td>1</td>\n",
       "      <td>0.000005</td>\n",
       "      <td>yes</td>\n",
       "      <td>Increased risk of nonsyndromic deafness</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>585</th>\n",
       "      <td>961</td>\n",
       "      <td>T</td>\n",
       "      <td>MT-RNR1</td>\n",
       "      <td>[\"T\",\"A\",\"G\"]</td>\n",
       "      <td>in-del</td>\n",
       "      <td>NaN</td>\n",
       "      <td>rRNA_gene</td>\n",
       "      <td>T961G</td>\n",
       "      <td>12S rRNA</td>\n",
       "      <td>no</td>\n",
       "      <td>yes</td>\n",
       "      <td>0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>1</td>\n",
       "      <td>0.000005</td>\n",
       "      <td>yes</td>\n",
       "      <td>Possibly DEAF-associated</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>587</th>\n",
       "      <td>961</td>\n",
       "      <td>T</td>\n",
       "      <td>MT-RNR1</td>\n",
       "      <td>[\"T\",\"C\"]</td>\n",
       "      <td>transition</td>\n",
       "      <td>NaN</td>\n",
       "      <td>rRNA_gene</td>\n",
       "      <td>T961C</td>\n",
       "      <td>12S rRNA</td>\n",
       "      <td>yes</td>\n",
       "      <td>yes</td>\n",
       "      <td>927</td>\n",
       "      <td>0.004730</td>\n",
       "      <td>139</td>\n",
       "      <td>0.000709</td>\n",
       "      <td>yes</td>\n",
       "      <td>DEAF, possibly LVNC-associated</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>590</th>\n",
       "      <td>961</td>\n",
       "      <td>T</td>\n",
       "      <td>MT-RNR1</td>\n",
       "      <td>[\"T\",\"C\"]</td>\n",
       "      <td>transition</td>\n",
       "      <td>NaN</td>\n",
       "      <td>rRNA_gene</td>\n",
       "      <td>T961TC</td>\n",
       "      <td>12S rRNA</td>\n",
       "      <td>yes</td>\n",
       "      <td>yes</td>\n",
       "      <td>927</td>\n",
       "      <td>0.004730</td>\n",
       "      <td>139</td>\n",
       "      <td>0.000709</td>\n",
       "      <td>yes</td>\n",
       "      <td>DEAF</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>593</th>\n",
       "      <td>961</td>\n",
       "      <td>T</td>\n",
       "      <td>MT-RNR1</td>\n",
       "      <td>[\"T\",\"G\"]</td>\n",
       "      <td>transversion</td>\n",
       "      <td>NaN</td>\n",
       "      <td>rRNA_gene</td>\n",
       "      <td>T961G</td>\n",
       "      <td>12S rRNA</td>\n",
       "      <td>yes</td>\n",
       "      <td>yes</td>\n",
       "      <td>1702</td>\n",
       "      <td>0.008684</td>\n",
       "      <td>1</td>\n",
       "      <td>0.000005</td>\n",
       "      <td>yes</td>\n",
       "      <td>Possibly DEAF-associated</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>595</th>\n",
       "      <td>961</td>\n",
       "      <td>T</td>\n",
       "      <td>MT-RNR1</td>\n",
       "      <td>[\"T\",\"TC\"]</td>\n",
       "      <td>in-del</td>\n",
       "      <td>NaN</td>\n",
       "      <td>rRNA_gene</td>\n",
       "      <td>T961C</td>\n",
       "      <td>12S rRNA</td>\n",
       "      <td>yes</td>\n",
       "      <td>yes</td>\n",
       "      <td>5</td>\n",
       "      <td>0.000025</td>\n",
       "      <td>2</td>\n",
       "      <td>0.000010</td>\n",
       "      <td>yes</td>\n",
       "      <td>DEAF, possibly LVNC-associated</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>598</th>\n",
       "      <td>961</td>\n",
       "      <td>T</td>\n",
       "      <td>MT-RNR1</td>\n",
       "      <td>[\"T\",\"TC\"]</td>\n",
       "      <td>in-del</td>\n",
       "      <td>NaN</td>\n",
       "      <td>rRNA_gene</td>\n",
       "      <td>T961TC</td>\n",
       "      <td>12S rRNA</td>\n",
       "      <td>yes</td>\n",
       "      <td>yes</td>\n",
       "      <td>5</td>\n",
       "      <td>0.000025</td>\n",
       "      <td>2</td>\n",
       "      <td>0.000010</td>\n",
       "      <td>yes</td>\n",
       "      <td>DEAF</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>599</th>\n",
       "      <td>961</td>\n",
       "      <td>T</td>\n",
       "      <td>MT-RNR1</td>\n",
       "      <td>[\"T\",\"TCCC\"]</td>\n",
       "      <td>in-del</td>\n",
       "      <td>NaN</td>\n",
       "      <td>rRNA_gene</td>\n",
       "      <td>T961C</td>\n",
       "      <td>12S rRNA</td>\n",
       "      <td>yes</td>\n",
       "      <td>no</td>\n",
       "      <td>1</td>\n",
       "      <td>0.000005</td>\n",
       "      <td>0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>yes</td>\n",
       "      <td>DEAF, possibly LVNC-associated</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>602</th>\n",
       "      <td>961</td>\n",
       "      <td>T</td>\n",
       "      <td>MT-RNR1</td>\n",
       "      <td>[\"T\",\"TCCC\"]</td>\n",
       "      <td>in-del</td>\n",
       "      <td>NaN</td>\n",
       "      <td>rRNA_gene</td>\n",
       "      <td>T961TC</td>\n",
       "      <td>12S rRNA</td>\n",
       "      <td>yes</td>\n",
       "      <td>no</td>\n",
       "      <td>1</td>\n",
       "      <td>0.000005</td>\n",
       "      <td>0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>yes</td>\n",
       "      <td>DEAF</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>603</th>\n",
       "      <td>961</td>\n",
       "      <td>T</td>\n",
       "      <td>MT-RNR1</td>\n",
       "      <td>[\"T\",\"TCCCCC\",\"TCCCCCC\"]</td>\n",
       "      <td>in-del</td>\n",
       "      <td>NaN</td>\n",
       "      <td>rRNA_gene</td>\n",
       "      <td>T961C</td>\n",
       "      <td>12S rRNA</td>\n",
       "      <td>no</td>\n",
       "      <td>yes</td>\n",
       "      <td>0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>1</td>\n",
       "      <td>0.000005</td>\n",
       "      <td>yes</td>\n",
       "      <td>DEAF, possibly LVNC-associated</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>606</th>\n",
       "      <td>961</td>\n",
       "      <td>T</td>\n",
       "      <td>MT-RNR1</td>\n",
       "      <td>[\"T\",\"TCCCCC\",\"TCCCCCC\"]</td>\n",
       "      <td>in-del</td>\n",
       "      <td>NaN</td>\n",
       "      <td>rRNA_gene</td>\n",
       "      <td>T961TC</td>\n",
       "      <td>12S rRNA</td>\n",
       "      <td>no</td>\n",
       "      <td>yes</td>\n",
       "      <td>0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>1</td>\n",
       "      <td>0.000005</td>\n",
       "      <td>yes</td>\n",
       "      <td>DEAF</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>620</th>\n",
       "      <td>1192</td>\n",
       "      <td>C</td>\n",
       "      <td>MT-RNR1</td>\n",
       "      <td>[\"C\",\"A\"]</td>\n",
       "      <td>transversion</td>\n",
       "      <td>NaN</td>\n",
       "      <td>rRNA_gene</td>\n",
       "      <td>C1192A</td>\n",
       "      <td>12S rRNA</td>\n",
       "      <td>yes</td>\n",
       "      <td>no</td>\n",
       "      <td>6</td>\n",
       "      <td>0.000031</td>\n",
       "      <td>0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>yes</td>\n",
       "      <td>DEAF-associated</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>623</th>\n",
       "      <td>1192</td>\n",
       "      <td>C</td>\n",
       "      <td>MT-RNR1</td>\n",
       "      <td>[\"C\",\"T\"]</td>\n",
       "      <td>transition</td>\n",
       "      <td>NaN</td>\n",
       "      <td>rRNA_gene</td>\n",
       "      <td>C1192T</td>\n",
       "      <td>12S rRNA</td>\n",
       "      <td>yes</td>\n",
       "      <td>yes</td>\n",
       "      <td>18</td>\n",
       "      <td>0.000092</td>\n",
       "      <td>3</td>\n",
       "      <td>0.000015</td>\n",
       "      <td>yes</td>\n",
       "      <td>DEAF-associated</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>640</th>\n",
       "      <td>1546</td>\n",
       "      <td>A</td>\n",
       "      <td>MT-RNR1</td>\n",
       "      <td>[\"A\",\"AT\"]</td>\n",
       "      <td>in-del</td>\n",
       "      <td>NaN</td>\n",
       "      <td>rRNA_gene</td>\n",
       "      <td>A1546T</td>\n",
       "      <td>12S rRNA</td>\n",
       "      <td>yes</td>\n",
       "      <td>no</td>\n",
       "      <td>9</td>\n",
       "      <td>0.000046</td>\n",
       "      <td>0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>yes</td>\n",
       "      <td>DEAF</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>641</th>\n",
       "      <td>1546</td>\n",
       "      <td>A</td>\n",
       "      <td>MT-RNR1</td>\n",
       "      <td>[\"A\",\"T\"]</td>\n",
       "      <td>transversion</td>\n",
       "      <td>NaN</td>\n",
       "      <td>rRNA_gene</td>\n",
       "      <td>A1546T</td>\n",
       "      <td>12S rRNA</td>\n",
       "      <td>no</td>\n",
       "      <td>yes</td>\n",
       "      <td>0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>1</td>\n",
       "      <td>0.000005</td>\n",
       "      <td>yes</td>\n",
       "      <td>DEAF</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>675</th>\n",
       "      <td>3243</td>\n",
       "      <td>A</td>\n",
       "      <td>MT-TL1</td>\n",
       "      <td>[\"A\",\"G\"]</td>\n",
       "      <td>transition</td>\n",
       "      <td>NaN</td>\n",
       "      <td>tRNA_gene</td>\n",
       "      <td>A3243G</td>\n",
       "      <td>tRNA Leu (UUR)</td>\n",
       "      <td>yes</td>\n",
       "      <td>yes</td>\n",
       "      <td>2</td>\n",
       "      <td>0.000010</td>\n",
       "      <td>49</td>\n",
       "      <td>0.000250</td>\n",
       "      <td>yes</td>\n",
       "      <td>MELAS / Leigh Syndrome / DMDF / MIDD / SNHL / ...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>678</th>\n",
       "      <td>3243</td>\n",
       "      <td>A</td>\n",
       "      <td>MT-TL1</td>\n",
       "      <td>[\"A\",\"T\"]</td>\n",
       "      <td>transversion</td>\n",
       "      <td>NaN</td>\n",
       "      <td>tRNA_gene</td>\n",
       "      <td>A3243T</td>\n",
       "      <td>tRNA Leu (UUR)</td>\n",
       "      <td>no</td>\n",
       "      <td>yes</td>\n",
       "      <td>0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>1</td>\n",
       "      <td>0.000005</td>\n",
       "      <td>yes</td>\n",
       "      <td>MM / MELAS / SNHL / CPEO</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>683</th>\n",
       "      <td>3254</td>\n",
       "      <td>C</td>\n",
       "      <td>MT-TL1</td>\n",
       "      <td>[\"C\",\"A\"]</td>\n",
       "      <td>transversion</td>\n",
       "      <td>NaN</td>\n",
       "      <td>tRNA_gene</td>\n",
       "      <td>C3254A</td>\n",
       "      <td>tRNA Leu (UUR)</td>\n",
       "      <td>yes</td>\n",
       "      <td>yes</td>\n",
       "      <td>81</td>\n",
       "      <td>0.000413</td>\n",
       "      <td>1</td>\n",
       "      <td>0.000005</td>\n",
       "      <td>yes</td>\n",
       "      <td>Gestational Diabetes (GDM)</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>688</th>\n",
       "      <td>3254</td>\n",
       "      <td>C</td>\n",
       "      <td>MT-TL1</td>\n",
       "      <td>[\"C\",\"T\"]</td>\n",
       "      <td>transition</td>\n",
       "      <td>NaN</td>\n",
       "      <td>tRNA_gene</td>\n",
       "      <td>C3254T</td>\n",
       "      <td>tRNA Leu (UUR)</td>\n",
       "      <td>yes</td>\n",
       "      <td>yes</td>\n",
       "      <td>66</td>\n",
       "      <td>0.000337</td>\n",
       "      <td>4</td>\n",
       "      <td>0.000020</td>\n",
       "      <td>yes</td>\n",
       "      <td>CPEO / poss. hypertension factor</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>693</th>\n",
       "      <td>3275</td>\n",
       "      <td>C</td>\n",
       "      <td>MT-TL1</td>\n",
       "      <td>[\"C\",\"A\"]</td>\n",
       "      <td>transversion</td>\n",
       "      <td>NaN</td>\n",
       "      <td>tRNA_gene</td>\n",
       "      <td>C3275A</td>\n",
       "      <td>tRNA Leu (UUR)</td>\n",
       "      <td>yes</td>\n",
       "      <td>no</td>\n",
       "      <td>20</td>\n",
       "      <td>0.000102</td>\n",
       "      <td>0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>yes</td>\n",
       "      <td>LHON</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>698</th>\n",
       "      <td>3275</td>\n",
       "      <td>C</td>\n",
       "      <td>MT-TL1</td>\n",
       "      <td>[\"C\",\"T\"]</td>\n",
       "      <td>transition</td>\n",
       "      <td>NaN</td>\n",
       "      <td>tRNA_gene</td>\n",
       "      <td>C3275T</td>\n",
       "      <td>tRNA Leu (UUR)</td>\n",
       "      <td>yes</td>\n",
       "      <td>no</td>\n",
       "      <td>6</td>\n",
       "      <td>0.000031</td>\n",
       "      <td>0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>yes</td>\n",
       "      <td>Metabolic syndrome and polycystic ovary syndrome</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>792</th>\n",
       "      <td>5794</td>\n",
       "      <td>T</td>\n",
       "      <td>MT-TC</td>\n",
       "      <td>[\"T\",\"C\"]</td>\n",
       "      <td>transition</td>\n",
       "      <td>NaN</td>\n",
       "      <td>tRNA_gene</td>\n",
       "      <td>T5794C</td>\n",
       "      <td>tRNA Cys</td>\n",
       "      <td>yes</td>\n",
       "      <td>yes</td>\n",
       "      <td>57</td>\n",
       "      <td>0.000291</td>\n",
       "      <td>4</td>\n",
       "      <td>0.000020</td>\n",
       "      <td>yes</td>\n",
       "      <td>Reported in tic disorder patient</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>793</th>\n",
       "      <td>5794</td>\n",
       "      <td>T</td>\n",
       "      <td>MT-TC</td>\n",
       "      <td>[\"T\",\"C\"]</td>\n",
       "      <td>transition</td>\n",
       "      <td>NaN</td>\n",
       "      <td>tRNA_gene</td>\n",
       "      <td>T5794C</td>\n",
       "      <td>tRNA Cys</td>\n",
       "      <td>yes</td>\n",
       "      <td>yes</td>\n",
       "      <td>57</td>\n",
       "      <td>0.000291</td>\n",
       "      <td>4</td>\n",
       "      <td>0.000020</td>\n",
       "      <td>yes</td>\n",
       "      <td>Hearing loss patient</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>795</th>\n",
       "      <td>5809</td>\n",
       "      <td>G</td>\n",
       "      <td>MT-TC</td>\n",
       "      <td>[\"G\",\"A\"]</td>\n",
       "      <td>transition</td>\n",
       "      <td>NaN</td>\n",
       "      <td>tRNA_gene</td>\n",
       "      <td>G5809A</td>\n",
       "      <td>tRNA Cys</td>\n",
       "      <td>yes</td>\n",
       "      <td>yes</td>\n",
       "      <td>16</td>\n",
       "      <td>0.000082</td>\n",
       "      <td>8</td>\n",
       "      <td>0.000041</td>\n",
       "      <td>yes</td>\n",
       "      <td>Hearing loss patient</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>796</th>\n",
       "      <td>5809</td>\n",
       "      <td>G</td>\n",
       "      <td>MT-TC</td>\n",
       "      <td>[\"G\",\"GA\"]</td>\n",
       "      <td>in-del</td>\n",
       "      <td>NaN</td>\n",
       "      <td>tRNA_gene</td>\n",
       "      <td>G5809A</td>\n",
       "      <td>tRNA Cys</td>\n",
       "      <td>yes</td>\n",
       "      <td>yes</td>\n",
       "      <td>1</td>\n",
       "      <td>0.000005</td>\n",
       "      <td>1</td>\n",
       "      <td>0.000005</td>\n",
       "      <td>yes</td>\n",
       "      <td>Hearing loss patient</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>809</th>\n",
       "      <td>7445</td>\n",
       "      <td>A</td>\n",
       "      <td>MT-CO1</td>\n",
       "      <td>[\"A\",\"C\"]</td>\n",
       "      <td>transversion</td>\n",
       "      <td>3</td>\n",
       "      <td>protein_coding_gene</td>\n",
       "      <td>A7445C</td>\n",
       "      <td>tRNA Ser (UCN) precursor</td>\n",
       "      <td>yes</td>\n",
       "      <td>no</td>\n",
       "      <td>1</td>\n",
       "      <td>0.000005</td>\n",
       "      <td>0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>yes</td>\n",
       "      <td>DEAF</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>813</th>\n",
       "      <td>7445</td>\n",
       "      <td>A</td>\n",
       "      <td>MT-CO1</td>\n",
       "      <td>[\"A\",\"G\"]</td>\n",
       "      <td>transition</td>\n",
       "      <td>3</td>\n",
       "      <td>protein_coding_gene</td>\n",
       "      <td>A7445G</td>\n",
       "      <td>tRNA Ser (UCN) precursor</td>\n",
       "      <td>no</td>\n",
       "      <td>yes</td>\n",
       "      <td>0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>6</td>\n",
       "      <td>0.000031</td>\n",
       "      <td>yes</td>\n",
       "      <td>SNHL</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>817</th>\n",
       "      <td>7445</td>\n",
       "      <td>A</td>\n",
       "      <td>MT-CO1</td>\n",
       "      <td>[\"A\",\"T\"]</td>\n",
       "      <td>transversion</td>\n",
       "      <td>3</td>\n",
       "      <td>protein_coding_gene</td>\n",
       "      <td>A7445T</td>\n",
       "      <td>tRNA Ser (UCN) precursor</td>\n",
       "      <td>yes</td>\n",
       "      <td>yes</td>\n",
       "      <td>4</td>\n",
       "      <td>0.000020</td>\n",
       "      <td>1</td>\n",
       "      <td>0.000005</td>\n",
       "      <td>yes</td>\n",
       "      <td>SNHL</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>984</th>\n",
       "      <td>16015</td>\n",
       "      <td>T</td>\n",
       "      <td>MT-TP</td>\n",
       "      <td>[\"T\",\"C\"]</td>\n",
       "      <td>transition</td>\n",
       "      <td>NaN</td>\n",
       "      <td>tRNA_gene</td>\n",
       "      <td>T16015C</td>\n",
       "      <td>tRNA Pro</td>\n",
       "      <td>no</td>\n",
       "      <td>yes</td>\n",
       "      <td>0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>3</td>\n",
       "      <td>0.000015</td>\n",
       "      <td>yes</td>\n",
       "      <td>Mitochondrial myopathy</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>985</th>\n",
       "      <td>16015</td>\n",
       "      <td>T</td>\n",
       "      <td>MT-TP</td>\n",
       "      <td>[\"T\",\"TATTCTCTGTTCTTTC\"]</td>\n",
       "      <td>in-del</td>\n",
       "      <td>NaN</td>\n",
       "      <td>tRNA_gene</td>\n",
       "      <td>T16015C</td>\n",
       "      <td>tRNA Pro</td>\n",
       "      <td>yes</td>\n",
       "      <td>no</td>\n",
       "      <td>2</td>\n",
       "      <td>0.000010</td>\n",
       "      <td>0</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>yes</td>\n",
       "      <td>Mitochondrial myopathy</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "      base wildtype     gene                   alleles      mutation  \\\n",
       "9      185        G   MT-CRb                 [\"G\",\"A\"]    transition   \n",
       "10     185        G   MT-CRb           [\"G\",\"A\",\"GAA\"]        in-del   \n",
       "16     195        T   MT-CRb             [\"T\",\"A\",\"C\"]        in-del   \n",
       "17     195        T   MT-CRb                 [\"T\",\"C\"]    transition   \n",
       "19     228        G   MT-CRb                 [\"G\",\"A\"]    transition   \n",
       "20     228        G   MT-CRb    [\"G\",\"A\",\"GACATAATAA\"]        in-del   \n",
       "38     576        A   MT-CRb                 [\"A\",\"G\"]    transition   \n",
       "40    3308        T   MT-ND1                 [\"T\",\"C\"]    transition   \n",
       "41    3308        T   MT-ND1                 [\"T\",\"G\"]  transversion   \n",
       "176   7445        A   MT-CO1                 [\"A\",\"C\"]  transversion   \n",
       "179   7445        A   MT-CO1                 [\"A\",\"G\"]    transition   \n",
       "266   8993        T  MT-ATP6                 [\"T\",\"C\"]    transition   \n",
       "269   8993        T  MT-ATP6                 [\"T\",\"G\"]  transversion   \n",
       "532    576        A   MT-CRb                 [\"A\",\"G\"]    transition   \n",
       "534    593        T    MT-TF                 [\"T\",\"C\"]    transition   \n",
       "536    593        T    MT-TF                [\"T\",\"TC\"]        in-del   \n",
       "557    745        A  MT-RNR1                [\"A\",\"AG\"]        in-del   \n",
       "559    745        A  MT-RNR1                 [\"A\",\"G\"]    transition   \n",
       "561    792        C  MT-RNR1                [\"C\",\"CT\"]        in-del   \n",
       "562    792        C  MT-RNR1                 [\"C\",\"T\"]    transition   \n",
       "585    961        T  MT-RNR1             [\"T\",\"A\",\"G\"]        in-del   \n",
       "587    961        T  MT-RNR1                 [\"T\",\"C\"]    transition   \n",
       "590    961        T  MT-RNR1                 [\"T\",\"C\"]    transition   \n",
       "593    961        T  MT-RNR1                 [\"T\",\"G\"]  transversion   \n",
       "595    961        T  MT-RNR1                [\"T\",\"TC\"]        in-del   \n",
       "598    961        T  MT-RNR1                [\"T\",\"TC\"]        in-del   \n",
       "599    961        T  MT-RNR1              [\"T\",\"TCCC\"]        in-del   \n",
       "602    961        T  MT-RNR1              [\"T\",\"TCCC\"]        in-del   \n",
       "603    961        T  MT-RNR1  [\"T\",\"TCCCCC\",\"TCCCCCC\"]        in-del   \n",
       "606    961        T  MT-RNR1  [\"T\",\"TCCCCC\",\"TCCCCCC\"]        in-del   \n",
       "620   1192        C  MT-RNR1                 [\"C\",\"A\"]  transversion   \n",
       "623   1192        C  MT-RNR1                 [\"C\",\"T\"]    transition   \n",
       "640   1546        A  MT-RNR1                [\"A\",\"AT\"]        in-del   \n",
       "641   1546        A  MT-RNR1                 [\"A\",\"T\"]  transversion   \n",
       "675   3243        A   MT-TL1                 [\"A\",\"G\"]    transition   \n",
       "678   3243        A   MT-TL1                 [\"A\",\"T\"]  transversion   \n",
       "683   3254        C   MT-TL1                 [\"C\",\"A\"]  transversion   \n",
       "688   3254        C   MT-TL1                 [\"C\",\"T\"]    transition   \n",
       "693   3275        C   MT-TL1                 [\"C\",\"A\"]  transversion   \n",
       "698   3275        C   MT-TL1                 [\"C\",\"T\"]    transition   \n",
       "792   5794        T    MT-TC                 [\"T\",\"C\"]    transition   \n",
       "793   5794        T    MT-TC                 [\"T\",\"C\"]    transition   \n",
       "795   5809        G    MT-TC                 [\"G\",\"A\"]    transition   \n",
       "796   5809        G    MT-TC                [\"G\",\"GA\"]        in-del   \n",
       "809   7445        A   MT-CO1                 [\"A\",\"C\"]  transversion   \n",
       "813   7445        A   MT-CO1                 [\"A\",\"G\"]    transition   \n",
       "817   7445        A   MT-CO1                 [\"A\",\"T\"]  transversion   \n",
       "984  16015        T    MT-TP                 [\"T\",\"C\"]    transition   \n",
       "985  16015        T    MT-TP  [\"T\",\"TATTCTCTGTTCTTTC\"]        in-del   \n",
       "\n",
       "    annotation        helix_feature MM_allele                 MM_feature  \\\n",
       "9          NaN           non_coding       G-A                  noncoding   \n",
       "10         NaN           non_coding       G-A                  noncoding   \n",
       "16         NaN           non_coding       T-C                  noncoding   \n",
       "17         NaN           non_coding       T-C                  noncoding   \n",
       "19         NaN           non_coding       G-A                  noncoding   \n",
       "20         NaN           non_coding       G-A                  noncoding   \n",
       "38         NaN           non_coding       A-G  noncoding MT-TF precursor   \n",
       "40           2  protein_coding_gene       T-C                        M1T   \n",
       "41           2  protein_coding_gene       T-G                     M1term   \n",
       "176          3  protein_coding_gene       A-C                   term514S   \n",
       "179          3  protein_coding_gene       A-G                term514term   \n",
       "266          2  protein_coding_gene       T-C                      L156P   \n",
       "269          2  protein_coding_gene       T-G                      L156R   \n",
       "532        NaN           non_coding     A576G  noncoding MT-TF precursor   \n",
       "534        NaN            tRNA_gene     T593C                   tRNA Phe   \n",
       "536        NaN            tRNA_gene     T593C                   tRNA Phe   \n",
       "557        NaN            rRNA_gene     A745G                   12S rRNA   \n",
       "559        NaN            rRNA_gene     A745G                   12S rRNA   \n",
       "561        NaN            rRNA_gene     C792T                   12S rRNA   \n",
       "562        NaN            rRNA_gene     C792T                   12S rRNA   \n",
       "585        NaN            rRNA_gene     T961G                   12S rRNA   \n",
       "587        NaN            rRNA_gene     T961C                   12S rRNA   \n",
       "590        NaN            rRNA_gene    T961TC                   12S rRNA   \n",
       "593        NaN            rRNA_gene     T961G                   12S rRNA   \n",
       "595        NaN            rRNA_gene     T961C                   12S rRNA   \n",
       "598        NaN            rRNA_gene    T961TC                   12S rRNA   \n",
       "599        NaN            rRNA_gene     T961C                   12S rRNA   \n",
       "602        NaN            rRNA_gene    T961TC                   12S rRNA   \n",
       "603        NaN            rRNA_gene     T961C                   12S rRNA   \n",
       "606        NaN            rRNA_gene    T961TC                   12S rRNA   \n",
       "620        NaN            rRNA_gene    C1192A                   12S rRNA   \n",
       "623        NaN            rRNA_gene    C1192T                   12S rRNA   \n",
       "640        NaN            rRNA_gene    A1546T                   12S rRNA   \n",
       "641        NaN            rRNA_gene    A1546T                   12S rRNA   \n",
       "675        NaN            tRNA_gene    A3243G             tRNA Leu (UUR)   \n",
       "678        NaN            tRNA_gene    A3243T             tRNA Leu (UUR)   \n",
       "683        NaN            tRNA_gene    C3254A             tRNA Leu (UUR)   \n",
       "688        NaN            tRNA_gene    C3254T             tRNA Leu (UUR)   \n",
       "693        NaN            tRNA_gene    C3275A             tRNA Leu (UUR)   \n",
       "698        NaN            tRNA_gene    C3275T             tRNA Leu (UUR)   \n",
       "792        NaN            tRNA_gene    T5794C                   tRNA Cys   \n",
       "793        NaN            tRNA_gene    T5794C                   tRNA Cys   \n",
       "795        NaN            tRNA_gene    G5809A                   tRNA Cys   \n",
       "796        NaN            tRNA_gene    G5809A                   tRNA Cys   \n",
       "809          3  protein_coding_gene    A7445C   tRNA Ser (UCN) precursor   \n",
       "813          3  protein_coding_gene    A7445G   tRNA Ser (UCN) precursor   \n",
       "817          3  protein_coding_gene    A7445T   tRNA Ser (UCN) precursor   \n",
       "984        NaN            tRNA_gene   T16015C                   tRNA Pro   \n",
       "985        NaN            tRNA_gene   T16015C                   tRNA Pro   \n",
       "\n",
       "    homoplasmic heteroplasmic  counts_hom    AF_hom  counts_het    AF_het  \\\n",
       "9           yes           yes       10568  0.053923         317  0.001617   \n",
       "10           no           yes           0  0.000000           1  0.000005   \n",
       "16           no           yes           0  0.000000           2  0.000010   \n",
       "17          yes           yes       34496  0.176015        1050  0.005358   \n",
       "19          yes           yes        9673  0.049356         191  0.000975   \n",
       "20           no           yes           0  0.000000           1  0.000005   \n",
       "38          yes           yes          10  0.000051           1  0.000005   \n",
       "40          yes           yes        1126  0.005745          26  0.000133   \n",
       "41          yes            no          89  0.000454           0  0.000000   \n",
       "176         yes            no           1  0.000005           0  0.000000   \n",
       "179          no           yes           0  0.000000           6  0.000031   \n",
       "266          no           yes           0  0.000000           2  0.000010   \n",
       "269          no           yes           0  0.000000           1  0.000005   \n",
       "532         yes           yes          10  0.000051           1  0.000005   \n",
       "534         yes           yes         480  0.002449          41  0.000209   \n",
       "536         yes            no          86  0.000439           0  0.000000   \n",
       "557          no           yes           0  0.000000           1  0.000005   \n",
       "559         yes           yes         333  0.001699           3  0.000015   \n",
       "561         yes            no           7  0.000036           0  0.000000   \n",
       "562         yes           yes          11  0.000056           1  0.000005   \n",
       "585          no           yes           0  0.000000           1  0.000005   \n",
       "587         yes           yes         927  0.004730         139  0.000709   \n",
       "590         yes           yes         927  0.004730         139  0.000709   \n",
       "593         yes           yes        1702  0.008684           1  0.000005   \n",
       "595         yes           yes           5  0.000025           2  0.000010   \n",
       "598         yes           yes           5  0.000025           2  0.000010   \n",
       "599         yes            no           1  0.000005           0  0.000000   \n",
       "602         yes            no           1  0.000005           0  0.000000   \n",
       "603          no           yes           0  0.000000           1  0.000005   \n",
       "606          no           yes           0  0.000000           1  0.000005   \n",
       "620         yes            no           6  0.000031           0  0.000000   \n",
       "623         yes           yes          18  0.000092           3  0.000015   \n",
       "640         yes            no           9  0.000046           0  0.000000   \n",
       "641          no           yes           0  0.000000           1  0.000005   \n",
       "675         yes           yes           2  0.000010          49  0.000250   \n",
       "678          no           yes           0  0.000000           1  0.000005   \n",
       "683         yes           yes          81  0.000413           1  0.000005   \n",
       "688         yes           yes          66  0.000337           4  0.000020   \n",
       "693         yes            no          20  0.000102           0  0.000000   \n",
       "698         yes            no           6  0.000031           0  0.000000   \n",
       "792         yes           yes          57  0.000291           4  0.000020   \n",
       "793         yes           yes          57  0.000291           4  0.000020   \n",
       "795         yes           yes          16  0.000082           8  0.000041   \n",
       "796         yes           yes           1  0.000005           1  0.000005   \n",
       "809         yes            no           1  0.000005           0  0.000000   \n",
       "813          no           yes           0  0.000000           6  0.000031   \n",
       "817         yes           yes           4  0.000020           1  0.000005   \n",
       "984          no           yes           0  0.000000           3  0.000015   \n",
       "985         yes            no           2  0.000010           0  0.000000   \n",
       "\n",
       "    pathogenic                                            disease  \n",
       "9          yes                                Low VO2max response  \n",
       "10         yes                                Low VO2max response  \n",
       "16         yes                       BD-associated / melanoma pts  \n",
       "17         yes                       BD-associated / melanoma pts  \n",
       "19         yes                                Low VO2max response  \n",
       "20         yes                                Low VO2max response  \n",
       "38         yes                               Hearing loss patient  \n",
       "40         yes  MELAS / DEAF enhancer / hypertension / LVNC / ...  \n",
       "41         yes                                Sudden Infant Death  \n",
       "176        yes                                               DEAF  \n",
       "179        yes                                               SNHL  \n",
       "266        yes                NARP / Leigh Disease / MILS / other  \n",
       "269        yes                NARP / Leigh Disease / MILS / other  \n",
       "532        yes                               Hearing loss patient  \n",
       "534        yes                          Nonsyndromic hearing loss  \n",
       "536        yes                          Nonsyndromic hearing loss  \n",
       "557        yes                                    DEAF-associated  \n",
       "559        yes                                    DEAF-associated  \n",
       "561        yes            Increased risk of nonsyndromic deafness  \n",
       "562        yes            Increased risk of nonsyndromic deafness  \n",
       "585        yes                           Possibly DEAF-associated  \n",
       "587        yes                     DEAF, possibly LVNC-associated  \n",
       "590        yes                                               DEAF  \n",
       "593        yes                           Possibly DEAF-associated  \n",
       "595        yes                     DEAF, possibly LVNC-associated  \n",
       "598        yes                                               DEAF  \n",
       "599        yes                     DEAF, possibly LVNC-associated  \n",
       "602        yes                                               DEAF  \n",
       "603        yes                     DEAF, possibly LVNC-associated  \n",
       "606        yes                                               DEAF  \n",
       "620        yes                                    DEAF-associated  \n",
       "623        yes                                    DEAF-associated  \n",
       "640        yes                                               DEAF  \n",
       "641        yes                                               DEAF  \n",
       "675        yes  MELAS / Leigh Syndrome / DMDF / MIDD / SNHL / ...  \n",
       "678        yes                           MM / MELAS / SNHL / CPEO  \n",
       "683        yes                         Gestational Diabetes (GDM)  \n",
       "688        yes                   CPEO / poss. hypertension factor  \n",
       "693        yes                                               LHON  \n",
       "698        yes   Metabolic syndrome and polycystic ovary syndrome  \n",
       "792        yes                   Reported in tic disorder patient  \n",
       "793        yes                               Hearing loss patient  \n",
       "795        yes                               Hearing loss patient  \n",
       "796        yes                               Hearing loss patient  \n",
       "809        yes                                               DEAF  \n",
       "813        yes                                               SNHL  \n",
       "817        yes                                               SNHL  \n",
       "984        yes                             Mitochondrial myopathy  \n",
       "985        yes                             Mitochondrial myopathy  "
      ]
     },
     "execution_count": 10,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "dups = uniqueOverlaps[uniqueOverlaps.duplicated(keep=False, subset=['base'])==True]\n",
    "dups"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.8.3"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 4
}
